tools/generate-helpers/pg-table-bindings/schema.go.tpl

```diff
@@ -38,6 +44,32 @@ import (
 {{- end}}
 
 var (
-   // {{template "createTableStmtVar" .Schema }} holds the create statement for table `{{.Schema.Table|upperCamelCase}}`.
-   {{template "createTableStmtVar" .Schema }} = {{template "createTableStmt" .Schema }}
+    // {{template "createTableStmtVar" .Schema }} holds the create statement for table `{{.Schema.Table|lowerCase}}`.
+    {{template "createTableStmtVar" .Schema }} = {{template "createTableStmt" .Schema }}
+
+    // {{template "schemaVar" .Schema}} is the go schema for table `{{.Schema.Table|lowerCase}}`.
+    {{template "schemaVar" .Schema}} = func() *walker.Schema {
+        schema := globaldb.GetSchemaForTable("{{.Schema.Table}}")
+        if schema != nil {
+            return schema
+        }
+        schema = walker.Walk(reflect.TypeOf(({{.Schema.Type}})(nil)), "{{.Schema.Table}}")
+        {{- /* Attach reference schemas, if provided. */ -}}
+        {{- $schema := .Schema }}
+        {{- range $idx, $ref := $schema.Parents}}
+            {{- if ne $ref.Table $schema.EmbeddedIn -}}.
+                WithReference({{template "schemaVar" $ref}})
+            {{- end }}
+        {{- end }}
+        {{- if .SearchCategory }}
+            {{- $ty := .Schema.Type }}
+            {{- /* special case */ -}}
+            {{- if eq $ty "*storage.Alert"}}
```

Comment:
Didn't we unify this so that we don't need to search ListAlert?

Reply:
Alert needs to be searchable appropriately because we will not be searching list alert as it's not saved

Reply:
I am just using the current mapping. Right now the options map for alerts is registered against `ListAlert`. https://github.com/stackrox/stackrox/blob/master/central/alert/mappings/options.go